### ISSUE TITLE
hyperparameter optimization uses test set instead of validation set

### ISSUE THREAD

**Issue by kevingreenman** [chemprop/chemprop#88]
hyperparameter_optimization uses cross_validate, which uses run_training. run_training returns `ensemble_scores`, which comes from doing evaluate_predictions on the test set.

I think doing hyperoptimization using scores on the validation set would be the correct way to do it. The documentation has a note hinting at this, but I think it leaves some ambiguity (when I read it, I thought it was just a caution to hold out your test set if you were trying to do hyperoptimization with cross-validation, since your test set would be changing each time). I think the default behavior and current documentation make it likely that a user will misuse the hyperoptimization functionality by not realizing they have to manually hold out their test set

**Comment by swansonk14** [comment]
Hi @kevingreenman, you're totally right, both about the current functionality and about the likelihood of causing confusion. The way we have used hyperparameter_optimization is by manually holding out an external test set and then letting hyperparameter_optimization split the remaining data into a train/val/test set, but users may not realize that they should do this. Do you think you could create a PR making the change from using the test set to using the validation set?

**Comment by kevingreenman** [comment]
@swansonk14 I gave it a shot in #90, let me know what you think

**Comment by kevingreenman** [comment]
@swansonk14 We're continuing to discuss the best way is to address this in the Green and Gomez-Bombarelli groups, it might be best to hold off on merging anything until we come to a more definitive conclusion

**Comment by swansonk14** [comment]
@kevingreenman Okay sounds good, just let me know whenever you want me to take a look

**Comment by hochej** [comment]
@swansonk14 @kevingreenman I came across this issue because I was also a bit confused as to why both a validation and a test set is required for hyperparameter optimization. So now I have a bit of a puzzle as to what data I should be providing the routine. If I split my data set (e.g. as part of an own nested CV routine) into train, val and test, then I will manually hold out the test set from the hyperoptimization. 
But I am not sure if I should either

- give the hyperoptimization the same val data for `separate_val_path` and `separate_test_path` 

or if it is necessary 

- to split off an additional part of train which is used for `separate_val_path` and that I use the val data for `separate_test_path`?

I hope it is clear what I mean and I would be grateful for hint!

**Comment by kevingreenman** [comment]
Hi @hochej, either approach is okay as long as your final test set is held out. `separate_val_path` is used for early stopping, while `separate_test_path` is used for selecting the best hyperparameters.

**Comment by hochej** [comment]
Thanks a lot!
